### PR TITLE
packages/salesforce: Update docs to reflect the use of optional security token

### PR DESCRIPTION
**Pull request by bacherol**
## Proposed commit message

It seems when SF instance demands a security token, it should be added to the password field.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices)

### CODE REVIEW

**Comment by shmsr** [comment]
@bacherol This is how README edits are done:

Make modifications to: `*/_dev/build/docs/README.md` and then run `elastic-package build` so that updated README gets generated for `docs/README.md`. If done the other i.e., changing `docs/README.md` and then running `elastic-package build` it'd result in no change to both as the reference is `*/_dev/build/docs/README.md` for `elastic-package build`.


I've also updated the field's description.

File: packages/salesforce/_dev/build/docs/README.md [hunk 1]
```diff
@@ -138,6 +138,13 @@ The User ID of the registered user.
 ### Password
 
 The password used to authenticate the user.
+If your Salesforce instance uses a security token, it should be appended at the end of the password without spaces or any characters.
+For example:
+```
+password = pass123
+security_token = sectoken321
+```
+This field should be filled as `pass123sectoken321`
```
Comment (shmsr) [inline packages/salesforce/_dev/build/docs/README.md:147]:
```suggestion

When using a Salesforce instance with a security token, append the token directly to your password without spaces or special characters. For example, if your password is `password` and your security token is `12345`, you would enter: `password12345`.
```

Comment (shmsr) [reply packages/salesforce/_dev/build/docs/README.md:147]: Suggestion to do it in a single line as the description is simple.

**Review by shmsr — APPROVED** [review]
I've added the CHANGELOG entry to go with 1.0.0 (GA). So, the changes won't be available until 8.16 stack release. If we 2 more PRs that are scheduled to go in with this 1.0.0 release of Salesforce; so added an entry there itself.

**Comment by shmsr** [comment]
@bacherol If the changes looks good; let's merge the PR.

**Review by kush-elastic — APPROVED** [review]
LGTM!

**Pull request merged**